src/data_type/mod.rs

```diff
@@ -2285,6 +2285,34 @@ impl DataType {
             (Err(_), Err(_)) => Err(Error::other("No common variant")),
         }
     }
+
+    pub fn lower_bound(&self) -> Option<DataType> {
+        match self {
+            DataType::Boolean(b) => b.max().map(|x| DataType::boolean_min(*x)),
```

Comment:
Why return an `Option` ?
You can simply return `DataType::Any` instead of None and the intersection will just work